r-pkg/tests/testthat/test-integration.R

```diff
@@ -474,5 +473,42 @@ ES_HOST <- "http://127.0.0.1:9200"
         expect_true(fieldDT[, length(unique(index))] >= 2)
     })
 
+#--- HTTP request helpers
+test_that(".request() works", {
+    testthat::skip_on_cran()
+    response <- uptasticsearch:::.request(
+        verb = "POST"
+        , url = "https://httpbin.org/status/201"
+        , body = NULL
+    )
+    expect_true(response$method == "POST")
+    expect_true(response$status_code == 201L)
+    expect_true(response$url == "https://httpbin.org/status/201")
+})
+
+test_that("retry logic works as expected", {
+    testthat::skip_on_cran()
+    futile.logger::flog.threshold(futile.logger::DEBUG)
+    log_lines <- testthat::capture_output({
+        response <- .request(
+            verb = "GET"
+            , url = "https://httpbin.org/status/502"
+            , body = NULL
```

Comment:
I forget, does one of the other integration tests use a body? If not maybe we should add one just for a lil more coverage

Reply:
Many do use it implicitly... for example `es_search()` is going to issue a `POST` request with the query attached as a body.

But agree it'd be good to have an explicit one here, to cover that `if()` block. I'll add that before merging.

Reply:
Just pushed https://github.com/uptake/uptasticsearch/pull/249/commits/51ee836553ce88f6efb2cebe296fc619c6db8aed with 2 more tests, explicitly covering the case where requests have a body.